<?xml version="1.0"?>
<launch>
  <arg name="bag_file"  default="$(env HOME)/storage/bags/2022-01-26/2022-01-26-18-11-14_valve_depth_throttled.bag"/>
  
  <include file="$(find object_keypoints_ros)/launch/keypoints.launch">
    <arg name="load_bag_filepath" default="false"/>
    <arg name="calibration_file"  default="$(env HOME)/calibration/intrinsics.yaml"/>
    <arg name="image_topic"       default="/hand_eye/color/image_raw_throttle"/>
    <arg name="camera_frame"      default="hand_eye_color_optical_frame"/>
    <arg name="model_name"        default="valve_depth"/>
    <arg name="keypoints_file"    default="$(find object_keypoints_ros)/config/valve.json"/>
    <arg name="image_file"        default=""/>
    <arg name="rviz_config"       default="$(find piloting_demo)/config/rviz/perception.rviz"/>
    <arg name="gpu"               default="false"/>
    <arg name="operation_mode"    default="service"/>
  </include>
  
  <node pkg="rosbag" type="play" name="rosbag_play" args="-r 0.1 $(arg bag_file)"/>

  <node name="perception" pkg="moma_mission" type="keypoints_perception.py" output="screen"/>
</launch>
